_core/regist/inc/sanitize.php

```diff
@@ -1,76 +1,79 @@
-        $resto = preg_replace("[\r\n]", "", $resto);
-        $com   = $this->CleanStr($com, $moderator); //But they can with this.
-
-        if (!$name || preg_match("/^[ |&#12288;|]*$/", $name))
-            $name = "";
-        if (!$com || preg_match("/^[ |&#12288;|\t]*$/", $com))
-            $com = "";
-        if (!$sub || preg_match("/^[ |&#12288;|]*$/", $sub))
-            $sub = "";
-
-        $name = str_replace(S_MANAGEMENT, '"' . S_MANAGEMENT . '"', $name);
-        $name = str_replace(S_DELETION, '"' . S_DELETION . '"', $name);
-
-        if (strlen($com) > S_POSTLENGTH)
-            return error(S_TOOLONG, $dest);
-        if (strlen($name) > 100)
-            return error(S_TOOLONG, $dest);
-        if (strlen($email) > 100)
-            return error(S_TOOLONG, $dest);
-        if (strlen($sub) > 100)
-            return error(S_TOOLONG, $dest);
-        if (strlen($resto) > 10)
-            return error(S_UNUSUAL, $dest);
-        if (strlen($url) > 10)
-            return error(S_UNUSUAL, $dest);
+    function process($post, $moderator = 0) {
+        $post['email'] = $this->CleanStr($post['email'], 0);
+        $post['email'] = preg_replace("[\r\n]", "", $post['email']);
+        $post['subject']   = $this->CleanStr($post['subject'], 0);
+        $post['subject']   = preg_replace("[\r\n]", "", $post['subject']);
+        /*$url   = $this->CleanStr($url, 0); //Or this
+        $url   = preg_replace("[\r\n]", "", $url);*/
+        $post['parent'] = $this->CleanStr($post['parent'], 0);
+        $post['parent'] = preg_replace("[\r\n]", "", $post['parent']);
+        $post['comment']   = $this->CleanStr($post['comment'], 1);
+
+        if (!$post['name'] || preg_match("/^[ |&#12288;|]*$/", $post['name']))
+            $post['name'] = "";
+        if (!$post['comment'] || preg_match("/^[ |&#12288;|\t]*$/", $post['comment']))
+            $post['comment'] = "";
+        if (!$post['subject'] || preg_match("/^[ |&#12288;|]*$/", $post['subject']))
+            $post['subject'] = "";
+
+        $post['name'] = str_replace(S_MANAGEMENT, '"' . S_MANAGEMENT . '"', $post['name']);
+        $post['name'] = str_replace(S_DELETION, '"' . S_DELETION . '"', $post['name']);
+
+        if (strlen($post['comment']) > S_POSTLENGTH)    return error(S_TOOLONG);
+        if (strlen($post['name']) > 100)                return error(S_TOOLONG);
+        if (strlen($post['email']) > 100)               return error(S_TOOLONG);
+        if (strlen($post['subject']) > 100)             return error(S_TOOLONG);
+        if (strlen($post['parent']) > 10)               return error(S_UNUSUAL);
+        //if (strlen($url) > 10)                        return error(S_UNUSUAL);
 
         // Standardize new character lines
-        $com = str_replace("\r\n", "\n", $com);
-        $com = str_replace("\r", "\n", $com);
+        $post['comment'] = str_replace("\r\n", "\n", $post['comment']);
+        $post['comment'] = str_replace("\r", "\n", $post['comment']);
```

Comment:
Potentially a regsub `/\r?\n/` (`(\r\n|\n)`), etc. whatever) to `\n`.

At most we should only have to remove `\r` and not `\n`.